mantidproject/mantid
# Workbench apply settings without relaunch

**Description of work.**

The algorithms category and interface hiding should now apply as soon as the setting dialog is closed.
The button has also been changed to be close rather than "SaveSettings", as most of the settings were saved even if you clicked the X close button.

Also harmonised both the close and X close actions

Also resolved a bug where the setting dialog reset the default instrument to the first in the facility list on starting the settings dialog.


**To test:**
1. open workbench
1. File->Settings
1. Fiddle with categories, both algorithms and interface hiding
1. check that the workbench toolboxes refresh when settings is closed.
1. try with both close and X close
1. Show invisible workspaces should also be immediate.

Now check that the default instrument is preserved.
1. In settings set a default instrument that is NOT the first in the list
1. close settings
1. exit workbench
1. restart workbench
1. open settings, check that the default instrument is still set
1. close and reopen settings, make sure it has not sneakily changed on you
1. close and reopen settings, change it to another instrument
1. close and reopen settings, make sure it has not sneakily changed on you



Fixes #27861 and fixes #27868 
 


The updating is in the release notes, the default instrument bug isn't as I think we introduced it in this release

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
